### PR TITLE
V7: Ensure session interface matches spec

### PR DESCRIPTION
**Pull request by bengourley**
The session class and usage required a bit of tweaking now that the spec has settled down.

### CODE REVIEW

File: packages/core/client.js [hunk 1]
```diff
@@ -139,11 +139,15 @@ class Client {
     const session = new Session()
 
     session.app = {
-      releaseStage: this._config.releaseStage,
-      version: this._config.appVersion,
-      type: this._config.appType
+      ...{
```
Comment (djskinner) [inline packages/core/client.js]: What does this creation and spreading of an object literal achieve?

File: packages/core/client.js [hunk 2]
```diff
@@ -139,11 +139,15 @@ class Client {
...
+        releaseStage: this._config.releaseStage,
+        version: this._config.appVersion,
+        type: this._config.appType
+      }
     }
 
+    session.user = { ...this._user }
```
Comment (djskinner) [inline packages/core/client.js]: Should this be `session._user`?

Comment (bengourley) [reply packages/core/client.js]: yes!

File: packages/core/client.js (hunk 1 above)
Comment (bengourley) [reply packages/core/client.js]: ah this a mistake, I'll rework it

**Pull request merged**